lib/route.js

```diff
@@ -271,9 +271,37 @@ function buildRouting (options) {
       // the route registration. To be sure to also load those hooks,
       // we must listen for the avvio's preReady event, and update the context object accordingly.
       avvio.once('preReady', () => {
-        for (const hook of lifecycleHooks) {
+        const hooks = {}
+
+        if (opts.hooks) {
```

Comment:
I'm still gonna change this property name to something fastify specific because of potential collisions, otherwise if some potential user with a custom plugin that uses a `hooks` property as a route-level option, his hooks might end up being skipped.